src/main/java/kr/motd/maven/os/Detector.java

```diff
@@ -143,4 +159,74 @@ private static String normalize(String value) {
         }
         return value.toLowerCase(Locale.US).replaceAll("[^a-z0-9]+", "");
     }
+
+    /**
+     * Gets a collection of OSes that the given detected OS is "like". The {@code detectedName} will
+     * always appear in the collection.
+     */
+    private static Collection<String> getOsLike(String detectedName) {
+        String[] osReleaseFileNames = {"/etc/os-release", "/usr/lib/os-release"};
+        Set<String> like = new LinkedHashSet<String>();
+        for (String osReleaseFileName : osReleaseFileNames) {
+            File file = new File(osReleaseFileName);
+            if (file.exists()) {
+                like.addAll(parseLinuxOsReleaseFile(file));
+                break;
+            }
+        }
+
+        // Older versions of redhat don't have /etc/os-release. In this case, just
+        // add "rhel" and "fedora" to the list.
+        if (like.isEmpty() && isRedHatVariant()) {
+            like.add("rhel");
+            like.add("fedora");
+        }
+
+        // Always add the detected name as well.
+        like.add(detectedName);
+
+        return like;
+    }
+
+    /**
+     * Indicates whether or not the OS is a variant of Redhat.
+     */
+    private static boolean isRedHatVariant() {
+        return new File("/etc/redhat-release").exists();
+    }
+
+    /**
+     * Parses a file in the format of {@code /etc/os-release} and returns a collection
+     * containing lower-case values taken from the {@code ID} and {@code ID_LIKE} entries.
+     */
+    private static Collection<String> parseLinuxOsReleaseFile(File file) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+
+            Set<String> likeSet = new LinkedHashSet<String>();
+            String line;
+            while((line = reader.readLine()) != null) {
+                // Parse the lines ID and ID_LIKE the same way.
+                int prefixLength = line.startsWith(ID_LIKE_PREFIX) ? ID_LIKE_PREFIX.length() :
+                    line.startsWith(ID_PREFIX) ? ID_PREFIX.length() : -1;
+                if (prefixLength < 0) {
+                    continue;
+                }
+
+                // Get the value after "=".
+                line = line.substring(prefixLength);
+
+                // Split the line on any whitespace.
+                String[] parts =  line.split("\\s+");
+                for (String part : parts) {
+                    likeSet.add(part.toLowerCase());
```

Comment:
The `toLowerCase()` here is harmful as-is. Either remove it because ID and ID_LIKE are required to already be lowercase, or specify a locale like Locale.ROOT.